src/main/java/tr/edu/ogu/ceng/controller/InternshipRegistryController.java

```diff
@@ -28,5 +33,20 @@ public ResponseEntity<InternshipRegistry> updateInternshipRegistry(@RequestBody
 	public ResponseEntity<InternshipRegistry> addInternshipRegistries(@RequestBody InternshipRegistry internshipRegistries) {
 		return ResponseEntity.ok(internshipRegistryService.addInternshipRegistry(internshipRegistries));
 	}
+    @GetMapping("/{userId}")
+    public ResponseEntity<Page<InternshipRegistry>> listAllInternshipRegistiries(@PathVariable Long userId,
+                                                                                        @RequestParam(defaultValue = "0") int page,
+                                                                                        @RequestParam(defaultValue = "10") int size,
+                                                                                        @RequestParam(defaultValue = "id") String sortBy) {
+        PageRequest pageRequest = PageRequest.of(page, size, Sort.by(sortBy));
+        Page<InternshipRegistry> internshipRegistries = internshipRegistryService.findAllRegistiriesByUserId(userId, pageRequest);
+        
+        if (internshipRegistries.isEmpty()) {
+            return ResponseEntity.noContent().build();
```

Comment:
Frontend tarafında aksiyona sebep olmaması için doğrudan boş liste de dönebiliriz

Reply:
güncelleme yaptım. inceleyebilirsiniz.